cmd/release/main.go

```diff
@@ -183,98 +222,101 @@ func run(root string, minisignPrivateKey string, dryRun bool) error {
+	sortPluginsByNameVersion(plugins)
+	return plugins, nil
+}
+
+func (c *command) loadMinisignPublicKeyFromFileOrPrivateKey(privateKey minisign.PrivateKey) (minisign.PublicKey, error) {
+	var publicKey minisign.PublicKey
+	if c.minisignPublicKey != "" {
+		var err error
+		publicKey, err = minisign.PublicKeyFromFile(c.minisignPublicKey)
+		if err != nil {
+			return minisign.PublicKey{}, err
+		}
+	} else if !privateKey.Equal(minisign.PrivateKey{}) {
+		var ok bool
+		publicKey, ok = privateKey.Public().(minisign.PublicKey)
+		if !ok {
+			return minisign.PublicKey{}, fmt.Errorf("unable to retrieve minisign public key from private key")
+		}
+	}
+	return publicKey, nil
+}
+
+func sortPluginsByNameVersion(plugins []release.PluginRelease) {
 	sort.Slice(plugins, func(i, j int) bool {
 		p1, p2 := plugins[i], plugins[j]
 		if p1.PluginName != p2.PluginName {
 			return p1.PluginName < p2.PluginName
 		}
 		return semver.Compare(p1.PluginVersion, p2.PluginVersion) < 0
 	})
-	if err := createPluginReleases(tmpDir, plugins); err != nil {
-		return fmt.Errorf("failed to create %s: %w", pluginReleasesFile, err)
-	}
+}
 
-	minisignPrivateKeyPassword := os.Getenv("MINISIGN_PRIVATE_KEY_PASSWORD")
-	publicKey, err := signPluginReleases(tmpDir, minisignPrivateKey, minisignPrivateKeyPassword)
+func (c *command) createRelease(ctx context.Context, client *release.Client, releaseName string, plugins []release.PluginRelease, tmpDir string, privateKey minisign.PrivateKey) error {
+	releaseBody, err := c.createReleaseBody(releaseName, plugins, privateKey)
 	if err != nil {
-		return fmt.Errorf("failed to sign %q: %w", filepath.Join(tmpDir, pluginReleasesFile), err)
-	}
-
-	if dryRun {
-		releaseBody := createReleaseBody(releaseName, plugins, publicKey)
-		if err := os.WriteFile(filepath.Join(tmpDir, "RELEASE.md"), []byte(releaseBody), 0644); err != nil { //nolint:gosec
-			return err
-		}
-		log.Printf("skipping GitHub release creation in dry-run mode")
-		log.Printf("release assets created in %q", tmpDir)
-		return nil
-	}
-	if err := createRelease(ctx, client, releaseName, plugins, tmpDir, publicKey); err != nil {
-		return fmt.Errorf("failed to create GitHub release: %w", err)
+		return err
 	}
-	return nil
-}
-
-func createRelease(ctx context.Context, client *github.Client, releaseName string, plugins []PluginRelease, tmpDir string, publicKey *minisign.PublicKey) error {
 	// Create GitHub release
-	release, _, err := client.Repositories.CreateRelease(ctx, githubReleaseOwner, githubRepo, &github.RepositoryRelease{
+	repositoryRelease, err := client.CreateRelease(ctx, c.githubReleaseOwner, release.GithubRepoPlugins, &github.RepositoryRelease{
 		TagName: github.String(releaseName),
 		Name:    github.String(releaseName),
-		Body:    github.String(createReleaseBody(releaseName, plugins, publicKey)),
+		Body:    github.String(releaseBody),
+		// Start release as a draft until all assets are uploaded
+		Draft: github.Bool(true),
```

Comment:
Tweaked the release process to mark it as a draft until all the plugins were uploaded, then change it to an official release at the end. I noticed our notifications of new releases were going out prior to all the assets being uploaded.